Comp2024/src/main/java/frc/robot/subsystems/LED.java

```diff
@@ -0,0 +1,145 @@
+    m_candle.configBrightnessScalar(0.7);
+
+    // Add options for colors in SmartDashboard
+    m_ledChooser.setDefaultOption("LED_Off", LEDColor.LEDCOLOR_OFF);
+    m_ledChooser.addOption("LED_White", LEDColor.LEDCOLOR_WHITE);
+    m_ledChooser.addOption("LED_Red", LEDColor.LEDCOLOR_RED);
+    m_ledChooser.addOption("LED_Orange", LEDColor.LEDCOLOR_ORANGE);
+    m_ledChooser.addOption("LED_Yellow", LEDColor.LEDCOLOR_YELLOW);
+    m_ledChooser.addOption("LED_Green", LEDColor.LEDCOLOR_GREEN);
+    m_ledChooser.addOption("LED_Blue", LEDColor.LEDCOLOR_BLUE);
+    m_ledChooser.addOption("LED_Purple", LEDColor.LEDCOLOR_PURPLE);
+
+    SmartDashboard.putData("LED_Color", m_ledChooser);
+    SmartDashboard.putBoolean("LED_normalMode", true);
+
+    initialize( );
+  }
+
+  @Override
+  public void periodic( )
+  {
+    // This method will be called once per scheduler run
+    setColor(m_ledChooser.getSelected( ));
+
+  }
+
+  @Override
+  public void simulationPeriodic( )
+  {
+
+  }
+
+  public void initialize( )
+  {
+    DataLogManager.log(String.format("%s: Subsystem initialized!", getSubsystem( )));
+
+  }
+
+  public void faultDump( )
+  {
+    DataLogManager.log(String.format("%s: faultDump  ----- DUMP FAULTS --------------", getSubsystem( )));
+  }
+
+  public void setColor(LEDColor color)
+  {
+    final String strName;
+
+    if (m_previousColor != color)
+    {
+      if (color == (LEDColor.LEDCOLOR_DASH))
+        color = m_ledChooser.getSelected( );
+      DataLogManager.log("");
+
+      switch (color)
+      {
+        default :
+        case LEDCOLOR_OFF :
+          strName = "OFF";
+          m_candle.setLEDs(0, 0, 0); // black
+          break;
+        case LEDCOLOR_WHITE :
+          strName = "WHITE";
+          m_candle.setLEDs(255, 255, 255); // white
+          break;
+        case LEDCOLOR_RED :
+          strName = "RED";
+          m_candle.setLEDs(255, 0, 0); // red
+          break;
+        case LEDCOLOR_ORANGE :
+          strName = "ORANGE";
+          m_candle.setLEDs(255, 80, 0); // orange
+          break;
+        case LEDCOLOR_YELLOW :
+          strName = "YELLOW";
+          m_candle.setLEDs(255, 255, 0); // yellow
+          break;
+        case LEDCOLOR_GREEN :
+          strName = "GREEN";
+          m_candle.setLEDs(0, 255, 0); // green
+          break;
+        case LEDCOLOR_BLUE :
+          strName = "BLUE";
+          m_candle.setLEDs(0, 0, 255); // blue
+          break;
+        case LEDCOLOR_PURPLE :
+          strName = "PURPLE";
+          m_candle.setLEDs(255, 0, 255); // purple
+          break;
+      }
+
+      DataLogManager.log(String.format("%s: color is now %s", getSubsystem( ), strName));
+      m_previousColor = color;
+
+    }
+  }
+
+  public void setNormalColor(LEDColor color)
+  {
+    Boolean normalMode = SmartDashboard.getBoolean("LED_normalMode", true);
+
+    if (normalMode)
+      setColor(color);
+  }
+
+  public void setLLColor(LEDColor color)

```

Comment:
I'm not exactly sure what normal mode is for, but it's likely different for 2024 and this method would not be needed. Need to investigate this one.